### ISSUE TITLE
Consider using absolute imports

### ISSUE THREAD

**Issue by cbrnr** [mne-tools/mne-python#12710]
I'd like to suggest to use absolute instead of relative imports. Absolute imports are more readable and in general provide better error messages, so PEP8 (and I 😄) recommend(s) using them (https://peps.python.org/pep-0008/#imports). Most of the work can probably be done by Ruff (https://docs.astral.sh/ruff/rules/relative-imports/).

**Comment by hoechenberger** [comment]
FWIW MNE-BIDS has been using absolute imports for many years, maybe @sappelhoff would care to share the reasoning behind and experience with that?

**Comment by cbrnr** [comment]
Although this is a separate issue, I'm also +💯 on switching to a src layout. I've also used both things (absolute imports and src layout) in MNELAB and SleepECG without any problems (not surprisingly, because these are the recommended approaches nowadays).

**Comment by agramfort** [comment]
>
> Although this is a separate issue, I'm also +💯 on switching to a src
> layout. I've also used both things (absolute imports and src layout) in
> MNELAB and SleepECG without any problems (not surprisingly, because these
> are the recommended approaches nowadays).
>
numpy, scipy, scikit-learn, joblib don't use src layout...

no objection with absolute imports

**Comment by hoechenberger** [comment]
Sorry about conflating this issue / request here, let's maybe not consider / discuss src layout for now and focus on absolute imports only :)

**Comment by sappelhoff** [comment]
> FWIW MNE-BIDS has been using absolute imports for many years, maybe @sappelhoff would care to share the reasoning behind and experience with that?

I have always tried to recommend / enforce this (when I had the decision power), for the same reasons that Clemens mentions. :-) I would be very happy if we adopted this in MNE-Python, too.

**Comment by hoechenberger** [comment]
> I have always tried to recommend / enforce this (when I had the decision power),

Yes, it was actually you who made me research this topic a little after you rejected my proposal to switch everything to relative imports a couple of years back or so :) this is when I figured that absolute imports are actually recommended.

**Comment by larsoner** [comment]
Okay with me to migrate to absolute imports. It'll probably create a lot of PR conflicts so we should make sure we don't have any big pending PRs open when we do it (or at least be aware of what they are so we can consciously decide to accept the rebase/merge pain)

**Comment by drammock** [comment]
> Okay with me to migrate to absolute imports.

no strong feeling. When doing so we should update our import/nesting tests to detect and fail if anyone puts in a relative import.
